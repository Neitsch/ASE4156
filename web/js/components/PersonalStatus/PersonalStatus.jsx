// @flow
import React from 'react';
import PropTypes from 'prop-types';
import Grid from 'material-ui/Grid';
import HighlightBox from '../HighlightBox';

import type { PersonalStatusRelay_bank }
  from './__generated__/PersonalStatusRelay_bank.graphql';

type Props = {
  bank: PersonalStatusRelay_bank,
}

const spacing = { xs: 12, sm: 6, md: 3 };

export default class PersonalStatus extends React.Component < Props > {
  static propTypes = {
    bank: PropTypes.shape({
      balance: PropTypes.number.isRequired,
      income: PropTypes.number.isRequired,
      outcome: PropTypes.number.isRequired,
    }).isRequired,
  }
  render() {
    return (
      <Grid container spacing={16} align="stretch" id="personal-status">
        {
          this.props.bank.balance == null
            ? null
            : (
              <Grid item {...spacing}>
                <HighlightBox title="Current balance" value={this.props.bank.balance.toFixed(2)} />
              </Grid>
            )
        }
        {this.props.bank.income == null ? null : (
          <Grid item {...spacing}>
            <HighlightBox title="Total Income" value={this.props.bank.income.toFixed(2)} />
          </Grid>
        )}
        {this.props.bank.outcome == null ? null : (
          <Grid item {...spacing}>
            <HighlightBox title="Total Expenditures" value={this.props.bank.outcome.toFixed(2)} />
          </Grid>
        )}
        {this.props.bank.income == null && this.props.bank.outcome == null
          ? null
          : (
            <Grid item {...spacing}>
              <HighlightBox
                title="Available money"
                value={(this.props.bank.income + this.props.bank.outcome).toFixed(2)}
              />
            </Grid>
          )}
      </Grid>
    );
  }
}
