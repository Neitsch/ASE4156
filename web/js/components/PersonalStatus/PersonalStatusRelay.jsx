// @flow
import { createFragmentContainer, graphql } from 'react-relay';
import PersonalStatus from './PersonalStatus';

export default createFragmentContainer(PersonalStatus, {
  bank: graphql`
    fragment PersonalStatusRelay_bank on GUserBank {
      balance
      income
      outcome
    }
` });
