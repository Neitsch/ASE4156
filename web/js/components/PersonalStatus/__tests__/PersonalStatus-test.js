/* @lazyspec (remove to manage manually) */
/* eslint-disable */
import PersonalStatus from '../PersonalStatus.jsx';

import React from 'react';
import { shallow } from 'enzyme';
import { shallowToJson } from 'enzyme-to-json';
import fakeProps from 'react-fake-props';

describe('PersonalStatus', () => {
  it('exists', () => {
    expect(PersonalStatus).toBeTruthy();
  });

  it('renders', () => {
    const props = fakeProps('web/js/components/PersonalStatus/PersonalStatus.jsx');
    console.log(props);
    const comp = <PersonalStatus {...props} />;
    const wrapper = shallow(comp);
    expect(shallowToJson(wrapper)).toMatchSnapshot();
  });
});
