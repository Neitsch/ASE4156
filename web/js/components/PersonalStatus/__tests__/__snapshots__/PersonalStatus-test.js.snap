// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PersonalStatus renders 1`] = `
<withStyles(GridWrapper)
  align="stretch"
  container={true}
  spacing={16}
>
  <withStyles(GridWrapper)
    item={true}
    sm={6}
    xs={12}
  >
    <HighlightBox
      title="Current balance"
    />
  </withStyles(GridWrapper)>
  <withStyles(GridWrapper)
    item={true}
    sm={6}
    xs={12}
  >
    <HighlightBox
      title="Total Income"
    />
  </withStyles(GridWrapper)>
  <withStyles(GridWrapper)
    item={true}
    sm={6}
    xs={12}
  >
    <HighlightBox
      title="Total Expenditures"
    />
  </withStyles(GridWrapper)>
  <withStyles(GridWrapper)
    item={true}
    sm={6}
    xs={12}
  >
    <HighlightBox
      title="Available money"
      value={NaN}
    />
  </withStyles(GridWrapper)>
</withStyles(GridWrapper)>
`;
